### DISCUSSION TITLE
API call returns object and I would like it to go into either a form or a text box not a table.

### DISCUSSION THREAD

**Discussion by JClarke90** [Budibase/budibase#11792] [Help] [answered]
I have an api query that returns an object not an array and I just want to display it withing text fields and not a table. is this possible?

So the object is like this

![image](https://github.com/Budibase/budibase/assets/12259768/effb1160-fafd-4474-adab-4d9fb2e575c3)

But I can't figure out how to get the name in either a heading or text box without using a table first. Is this possible?

![image](https://github.com/Budibase/budibase/assets/12259768/22922f88-a295-46d9-bba0-c06c7198c226)

**Comment by ConorWebb96** [comment, marked as answer]
Hey @JClarke90,

This is something which can be done. I've setup a very simple rest request below that returns only 1 object. Below ill show how you can access this object and bind it up components so it can be displayed.

Heres my rest output;
```
{
  "employeeID": "E12345",
  "firstName": "John",
  "lastName": "Doe",
  "age": 30,
  "isEmployed": true,
  "skills": ["JavaScript", "Python", "SQL"],
  "address": {
    "street": "123 Main Street",
    "city": "Anytown",
    "state": "CA",
    "zipCode": "12345"
  },
  "manager": null,
  "hourlyRate": 25.5,
  "hireDate": "2023-09-18T09:00:00Z",
  "jobTitle": "Software Developer",
  "isFullTime": true,
  "website": "https://www.example.com/johndoe",
  "phoneNumber": "555-123-4567",
  "employeeSince": 1591234567
}
```
<img width="953" alt="Screenshot 2023-09-18 at 09 06 32" src="https://github.com/Budibase/budibase/assets/126772285/6cc2dca6-3cb8-4c38-9eab-2c7227dbf0f4">

As the screenshot you provided uses a form, I will do the same for the below;

1. Setup your form component
2. Setup your data provided nested within your form
3. Add your form field components
4. You can bind the values generated from your rest query by hooking into the data provider.

In my case I will display the Full name using the first and last name `{{ New Data Provider.Rows.0.firstName }} {{ New Data Provider.Rows.0.lastName }}`

<img width="1393" alt="Screenshot 2023-09-18 at 09 12 31" src="https://github.com/Budibase/budibase/assets/126772285/135e1c16-873c-4467-8bc5-deaa462c0241">

I've also included the age as well as an additional field; `{{ New Data Provider.Rows.0.age }}`.

In your case it would be the `data provider.Rows.0.name`

I hope this helps.

**Comment by JClarke90** [comment]
That working perfectly. Thank you.
